### PR TITLE
Fix OSX and MXE Windows builds

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

SUMMARY: None

#### Purpose of change

Follow-up after #31361.

### CODE REVIEW

File: src/mingw.thread.h [hunk 1]
```diff
@@ -64,7 +64,8 @@ class thread
             using Call = decltype( std::bind( f, args... ) );
             Call *call = new Call( std::bind( f, args... ) );
             mHandle = ( HANDLE )_beginthreadex( NULL, 0, threadfunc<Call>,
-                                                ( LPVOID )call, 0, ( unsigned * ) & ( mThreadId.mId ) );
+                                                static_cast<LPVOID>( call ),
+                                                0, ( unsigned * ) & ( mThreadId.mId ) );
```
Comment (narc0tiq) [inline src/mingw.thread.h]: What about this `unsigned *`, is this not also a cast?

Comment (ZhilkinSerg) [inline src/mingw.thread.h]:
```suggestion
                                                0, reinterpret_cast<unsigned *>( & ( mThreadId.mId ) ) );
```

Comment (ZhilkinSerg) [reply src/mingw.thread.h]: Yes, there are maybe a few more casts - which I could miss using web UI.

**Comment by ZhilkinSerg** [comment]
I think we are good here. Self merging to see if it helped or we need some more fixes.

**Comment by jbytheway** [comment]
Thanks for following up :).

**Pull request merged**